tests/program-config.test.ts

```diff
@@ -0,0 +1,172 @@
+import { test } from 'node:test';
+import assert from 'node:assert';
+import { Bytes, Field } from 'o1js';
+import {
+  Spec,
+  Input,
+  Operation,
+  Node,
+  Attestation,
+  type UserInputs,
+  splitUserInputs,
+  recombineDataInputs,
+} from '../src/program-config.ts';
+import { createAttestation } from './test-utils.ts';
+
+test(' Spec and Node operations', async (t) => {
+  const Bytes32 = Bytes(32);
+
+  await t.test('Basic Spec with equality check', () => {
+    const InputData = { age: Field };
+    const spec = Spec(
+      {
+        data: Input.private(InputData),
+        targetAge: Input.public(Field),
+      },
+      ({ data, targetAge }) => ({
+        assert: Operation.equals(Operation.property(data, 'age'), targetAge),
+        data: Operation.property(data, 'age'),
+      })
+    );
+
+    const root = {
+      data: { age: Field(25) },
+      targetAge: Field(25),
+    };
+
+    const assertResult = Node.eval(root, spec.logic.assert);
+    const dataResult = Node.eval(root, spec.logic.data);
+
+    assert.strictEqual(assertResult.toString(), 'true');
```

Comment:
nit: can you use `assertResult.toBoolean()` here and compare with `true`, that seems more idiomatic for bool outputs

(same in the other bool tests)